src/Time/Formatting.hs

```diff
@@ -63,7 +66,9 @@ instance (KnownRatName unit, Series units)
                     timeStr = case floor newUnit :: Int of
                                    0 -> ""
                                    _ -> show format
-                in timeStr ++ seriesF @units @unit (newUnit - format)
+                in if t == 0 || format == newUnit
+                   then show newUnit
+                   else timeStr ++ seriesF @units @unit (newUnit - format)
```

Comment:
I have a feeling that this could be written simpler... I think that

* `floor newUnit == 0 ≡ floorUnit newUnit == 0`

So at least one call to `floor` is redundant. Also, I think that `format` is not a good name for variable with value `floorUnit newUnit`...